testing/bdd/features/styles/ebuttd_style_references.feature

```diff
@@ -0,0 +1,85 @@
+Feature: Remove style elements that refer to other style elements
+
+  Examples:
+    | xml_file                     |
+    | style_element_references.xml |
+
+
+  Scenario: Convert style element with reference to another style to a presentationally equivalent one
+    Given an xml file <xml_file>
+    When  it contains style "s1"
+    And   style "s1" has attribute "color" set to "white"
+    And   style "s1" has attribute "backgroundColor" set to "black"
+    And   it contains style "s2"
+    And   style "s2" has attribute "style" set to "s1"
+    And   style "s2" has attribute "fontFamily" set to "reith"
+    And   it contains style "s3"
+    And   style "s3" has attribute "fontFamily" set to "monospace"
+    And   it contains style "s4"
+    And   style "s4" has attribute "style" set to "s2 s3"
+    And   style "s4" has attribute "color" set to "yellow"
+    And   it contains some text with style "s4"
+    When  the document is generated
+    And   the EBU-TT-Live document is converted to EBU-TT-D
+    Then  the ebu_tt_d document contains style "s4" with attribute "fontFamily" set to "monospace"
+    And   the ebu_tt_d document contains style "s4" with attribute "color" set to "#ffff00ff"
+    And   the ebu_tt_d document contains style "s4" with attribute "backgroundColor" set to "#000000ff"
+
+
+  Scenario: Convert padding specifed on style applied to region to padding specified only on region when padding is on both style and region
+    Given an xml file <xml_file>
+    When  it contains style "s1"
+    And   style "s1" has attribute "padding" set to "10px"
+    And   it contains region "r1"
+    And   region "r1" has attribute "style" set to "s1"
+    And   region "r1" has attribute "padding" set to "5px"
+    And   it contains some text with style "s1"
```

Comment:
OK, do you mean the input (EBUTT3) document needs to contain a `region` attribute at the `<p>` level? I guess it doesn't really make sense at the `<span>` level.
i.e.
```xml
<tt:div>
  <tt:p begin="500ms" end="3420ms" region="r1" xml:id="ID006">
    <tt:span>Some example text...</tt:span>
  </tt:p>
</tt:div>
```

Reply:
The example above would work, yes. It would also be valid to reference the region from a different content element, i.e. `body`, `div` or `span` but I don't think it's important to test all those possibilities.